pkg/volume/cinder/attacher.go

```diff
@@ -278,12 +279,22 @@ func (detacher *cinderDiskDetacher) Detach(deviceMountPath string, nodeName type
 		instanceid = instanceid[(ind + 1):]
 	}
 
+	pending, volumeStatus, err := detacher.cinderProvider.OperationPending(volumeID)
+	if err != nil {
+		return err
+	}
+	if pending {
+		glog.Infof("status of volume %q is %s, wait it to be finished", volumeID, volumeStatus)
+		return volume.ErrNotFinish
```

Comment:
I think that you should wait for the operation to finish here (with some sane timeout). You should not mis-use attach/detach controller loop to ignore some errors